sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -528,11 +530,22 @@ abstract class UnresolvedStarBase(target: Option[Seq[String]]) extends Star with
     // (i.e. [name].* is both a table and a struct), the struct path can always be qualified.
     val attribute = resolve(target.get, resolver)
     if (attribute.isDefined) {
+      // If cleanupNestedAliasesDuringStructExpansion is true, we remove nested aliases during
+      // struct expansion. This is something which is done in the CleanupAliases rule but for the
+      // single-pass analyzer it has to be done here to avoid additional tree traversals.
```

Comment:
Seems we can always do it? It's no harm to clean up nested alias earlier.

Reply:
Probably. But this PR is only introduces new functionality or changes that don't have any effect on the existing code. If we want to change existing Catalyst behavior, that needs to be done in separate PR.

Reply:
We probably can't do that without breaking current behavior. I've encountered multiple examples where nested `Aliases` are actually used in computing the name of the outer `Alias`. Removing inner `Alias` before computing the outer one will result in schema changes